spyderlib/widgets/calltip.py

```diff
@@ -211,7 +211,7 @@ def show_tip(self, point, tip, wrapped_tiplines):
                     vertical = 'top'
             else:
                 vertical = 'top'
-        if point.x() + tip_width > screen_rect.width():
+        if point.x() + tip_width > screen_rect.width() + screen_rect:
```

Comment:
I think you are missing here the x coordinate, like this:

    screen_rect.x()

At least that's what you have in the IPython commit you referenced